crypto/asn1/asn_mime.c

```diff
@@ -663,7 +663,10 @@ static STACK_OF(MIME_HEADER) *mime_parse_hdr(BIO *bio)
                     mime_debug("Found End Value\n");
                     *p = 0;
                     mhdr = mime_hdr_new(ntmp, strip_ends(q));
-                    sk_MIME_HEADER_push(headers, mhdr);
+                    if (mhdr == NULL)
+                        goto err;
+                    if (!sk_MIME_HEADER_push(headers, mhdr))
+                        goto err;
```

Comment:
We can't set `mhdr` to NULL, as it is reuse just after the `for` loop ending:
...
`} else if (state == MIME_VALUE) `
`    mime_hdr_addparam(mhdr, ntmp, strip_ends(q));`
...
I do another change to avoid this double free.